### ISSUE TITLE
Cannot set stderr prefix/postfix; KeyError: 'rightred'

### ISSUE THREAD

**Issue by t184256** [xonsh/xonsh#2778]
I cannot set `$XONSH_STDERR_PREFIX` and `$XONSH_STDERR_POSTFIX` without xonsh crashing.

## xonfig

```                    
+------------------+-----------------+
| xonsh            | 0.7.5           |
| Python           | 3.6.6           |
| PLY              | 3.7             |
| have readline    | True            |
| prompt toolkit   | 2.0.4           |
| shell type       | prompt_toolkit2 |
| pygments         | 2.2.0           |
| on posix         | True            |
| on linux         | True            |
| distro           | unknown         |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

## Expected Behavior
stderr output gets styled according to prefix and postfix

## Current Behavior
xonsh crashes as soon as I set both:

```
$ XONSH_DEBUG=1 xonsh --no-rc
...
monk@cola ~ $ $XONSH_STDERR_POSTFIX = 'a'                                       
monk@cola ~ $ $XONSH_STDERR_PREFIX = 'a'                                        
Traceback (most recent call last):
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/main.py", line 343, in main
    return main_xonsh(args)
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/main.py", line 369, in main_xonsh
    shell.shell.cmdloop()
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/ptk2/shell.py", line 170, in cmdloop
    self.default(line)
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/base_shell.py", line 327, in default
    tee = Tee(encoding=enc, errors=err)
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/base_shell.py", line 237, in __init__
    poststderr = format_std_prepost(env.get('XONSH_STDERR_POSTFIX'))
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/tools.py", line 1792, in format_std_prepost
    s = shell.format_color(invis + s + invis, force_string=True)
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/ptk2/shell.py", line 265, in format_color
    formatter = pyghooks.XonshTerminal256Formatter(style=proxy_style)
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/lazyasd.py", line 85, in __call__
    return obj(*args, **kwargs)
  File "/nix/store/kgibmzwk0bgr3hnfdckmsbwg2ajpyb7x-xonsh-0.7.5/lib/python3.6/site-packages/xonsh/pyghooks.py", line 1350, in __init__
    super().__init__(*args, **kwargs)
  File "/nix/store/w7p41gs88mkpk8k0bwbgg7bc3970fpbl-python3.6-Pygments-2.2.0/lib/python3.6/site-packages/pygments/formatters/terminal256.py", line 134, in __init__
    self._setup_styles()  # convert selected style's colors to term. colors
  File "/nix/store/w7p41gs88mkpk8k0bwbgg7bc3970fpbl-python3.6-Pygments-2.2.0/lib/python3.6/site-packages/pygments/formatters/terminal256.py", line 224, in _setup_styles
    self.style_string[str(ttype)] = (escape.color_string(),
  File "/nix/store/w7p41gs88mkpk8k0bwbgg7bc3970fpbl-python3.6-Pygments-2.2.0/lib/python3.6/site-packages/pygments/formatters/terminal256.py", line 53, in color_string
    esc = codes[self.fg[5:]]
KeyError: 'rightblue'
Xonsh encountered an issue during launch
Failback to /run/current-system/sw/bin/bash
```

The particular truncated color name varies from run to run.

## Steps to Reproduce

```
$XONSH_STDERR_POSTFIX = 'a'                                       
$XONSH_STDERR_PREFIX = 'a'  
```

**Comment by t184256** [comment]
Works as expected with ptk 1.0.15 though.

**Comment by melund** [comment]
Thanks for reporting @t184256. I think I know why this happens. With PTK2 the ansi-color names changed. Xonsh follows the new names and convert them back when using the readline and PTK1 backend. I think I overlooked a place where xonsh does coloring without going through PTK2 and the colors are therefore not converted back.  That causes a problem with pygments which doesn't understand the new names. 

We have put in a PR to update pygments, but that will take a while. We can probably make a workaround in xonsh in the meantime. 

I will try to fix this when I get a little more time later this week.

**Comment by melund** [comment]
@t184256 I have a potential fix in #2783

**Comment by t184256** [comment]
Well, it doesn't crash now and seems to prepend/append the prefix/postfix to stderr now. Twice.

Thanks for bothering to hack around it right now, but waiting for new pygments with your patch seems saner.
